src/kademlia/get_item.cpp

```diff
@@ -70,6 +70,14 @@ void get_item::got_data(bdecode_node const& v,
 		{
 			if (!m_data.assign(v, salt, seq, pk, sig))
 				return;
+
+			// For get_item, we should call callback when we get data.
+			// The data maybe is not the latest, but we can update it later.
+			// Waitting transaction timeout (15 seconds) is not efficient.
+			// For put_item, it put new values to DHT when done(), so this
+			// change shouldn't break anything since put_item callback
+			// almost doing nothing, just a notification.
+			m_data_callback(m_data);
```

Comment:
does this cause the callback to be called more than once? (or was that the case previously?). I would just have expected that you'd remove a call to it from somewhere also (and maybe protect this call to only be made once).
I suppose it's not that big of a deal to call it multiple times, it can be a bit difficult to use though. The ideal semantics would be that it's always called exactly once, even if there is no response.

Speaking of which, don't you think it would be useful to know whether the call was "early" or "authoritative"? i.e. whether it's the first response or the most recent version of the data (i.e. the last response).